src/Components/CardboardModal/CardboardModal.tsx

```diff
@@ -0,0 +1,129 @@
+        subTitle,
+        titleIconName
+    } = props;
+
+    // contexts
+
+    // state
+
+    // hooks
+    const { t } = useTranslation();
+    const titleId = useId('modal-title');
+
+    // callbacks
+
+    // side effects
+
+    // styles
+    const classNames = getClassNames(styles, {
+        theme: useTheme(),
+        isDestructiveFooterActionVisible: !!destructiveButtonProps
+    });
+
+    return (
+        <Modal
+            isOpen={isOpen}
+            titleAriaId={titleId}
+            onDismiss={onDismiss}
+            styles={classNames.subComponentStyles.modal}
+        >
+            <Stack tokens={stackTokens} style={{ height: '100%' }}>
+                <div className={classNames.headerContainer}>
+                    <div className={classNames.titleContainer}>
+                        {titleIconName && (
+                            <Icon
+                                iconName={titleIconName}
+                                styles={classNames.subComponentStyles.icon}
+                            />
+                        )}
+                        <h3 id={titleId} className={classNames.title}>
+                            {typeof title === 'function' ? title() : title}
+                        </h3>
+                    </div>
+                    {subTitle && (
+                        <p className={classNames.subtitle}>
+                            {typeof subTitle === 'function'
+                                ? subTitle()
+                                : subTitle}
+                        </p>
+                    )}
+                </div>
+                <div className={classNames.content}>
+                    <Stack tokens={stackTokens}>{children}</Stack>
+                </div>
+                <div className={classNames.footer}>
+                    <Stack
+                        horizontal
+                        tokens={stackTokens}
+                        styles={classNames.subComponentStyles.footerStack}
+                    >
+                        {destructiveButtonProps && (
+                            <PrimaryButton
+                                {...destructiveButtonProps}
+                                styles={classNames.subComponentStyles.destructiveButton?.()}
+                            />
+                        )}
+                        <Stack
+                            tokens={stackTokens}
+                            horizontal
+                            horizontalAlign={'end'}
+                        >
+                            <PrimaryButton
+                                {...primaryButtonProps}
+                                styles={classNames.subComponentStyles.primaryButton?.()}
+                            />
+                            <DefaultButton
+                                text={t('cancel')}
+                                onClick={onDismiss}
+                                styles={classNames.subComponentStyles.cancelButton?.()}
```

Comment:
do consumer have to provide styling for it (seems it is optional as subcomponent style), shouldn't we have default styling for secondary actions (like cancel)?

Reply:
we do have default styling. it comes from the styles file. If they want to override, they have the option to, but the defautl is to have these styles here.

Reply:
Made the danger button non-optional since we always supply that.